edge-functions/ab-testing-statsig/pages/_middleware.ts

```diff
@@ -0,0 +1,53 @@
+import { NextRequest, NextResponse } from 'next/server'
+import statsig from 'statsig-node'
+
+// Store a cookie for the user
+const UID_COOKIE = 'uid'
+
+export async function middleware(req: NextRequest) {
+  // Clone the URL
+  const url = req.nextUrl.clone()
+
+  // Prevent users from access buckets directly
+  if (url.pathname.startsWith(`/_bucket`)) {
+    url.pathname = '/404'
+
+    return NextResponse.rewrite(url)
+  }
+
+  // Just run for the / path
+  if (req.nextUrl.pathname !== '/') {
+    return NextResponse.next()
+  }
+
+  // Initialize statsig client
+  await statsig.initialize(process.env.STATSIG_SERVER_API_KEY as string)
+
+  // Get users UID from the cookie
+  let userID = req.cookies[UID_COOKIE]
+
+  // Set a userID if not present
+  if (!userID) {
+    userID = crypto.randomUUID()
+  }
+
+  // Fetch experiment
+  const experiment = await statsig.getExperiment({ userID }, 'half_bucket')
```

Comment:
Is this doing caching? if so can we mention it, or in the place where it's happening, either here or by calling `statsig.initialize` or in memory after it's imported, either way we should mention that to make it easier to understand why is latency not being affected by this (from my own testing).